libpages/config/config_v1.go

```diff
@@ -92,13 +106,43 @@ func (c *V1) Version() Version {
 	return Version1
 }
 
+func (c *V1) getCachedCleartextPassword(username string) (cleartext string, ok bool) {
+	c.cleartextPasswordCacheLock.RLock()
+	defer c.cleartextPasswordCacheLock.RUnlock()
+	cleartext, ok = c.cleartextPasswordCache[username]
+	return cleartext, ok
+}
+
+func (c *V1) cacheCleartextPassword(username string, cleartext string) {
+	c.cleartextPasswordCacheLock.Lock()
+	defer c.cleartextPasswordCacheLock.Unlock()
+	c.cleartextPasswordCache[username] = cleartext
+}
+
 // Authenticate implements the Config interface.
-func (c *V1) Authenticate(username, password string) bool {
+func (c *V1) Authenticate(ctx context.Context, username, cleartextPassword string) bool {
+	if c.EnsureInit() != nil {
+		return false
+	}
+
+	if cleartext, ok := c.getCachedCleartextPassword(username); ok {
+		return cleartext == cleartextPassword
```

Comment:
Shouldn't the `false` case here actually be treated as a cache miss, and so we should bcrypt it?

Reply:
What if the password changes?

Reply:
The entire config object gets blown away when it expires, so bcrypt will run again. It's currently 16s but I'm thinking increase it to 2m.